src/main/scala/firrtl/Emitter.scala

```diff
@@ -543,7 +543,13 @@ class VerilogEmitter extends Emitter {
                   val rmem_port = WSubAccess(mem,raddrx,s.data_type,UNKNOWNGENDER)
                   assign(rdata,rmem_port)
                   val wmem_port = WSubAccess(mem,waddrx,s.data_type,UNKNOWNGENDER)
-                  update(wmem_port,datax,clk,AND(AND(enx,maskx),wmode))
+
+                  val tempName = namespace.newTemp
+                  val tempExp = AND(enx,maskx)
+                  declare("wire", tempName, tpe(tempExp))
+                  val tempWRef = wref(tempName, tpe(tempExp))
+                  assign(tempWRef, tempExp)
+                  update(wmem_port,datax,clk,AND(tempWRef,wmode))
```

Comment:
I'm curious how this could be moved to another pass. This essentially creates logic that guards the writing of a rw port to be the AND of write enable, write mask, and write mode. According to FIRRTL semantics, these are 3 separate fields that are only being merged here to simulate in Verilog. 

I guess there could be a pass on LoFirrtl that replaces FIRRTL mems with some VerilogSimMem or something that merges these three signals into a single one?

A reasonable alternative is to go back to `AND(AND(enx, maskx), wmode)` and instead just make the emitter handle that since it's a simple logical condition for writing to the port. The main issue with how it is currently implemented is that this update function is using the same emission function used to emit all primops (which disallows nesting) when instead it could use a special one that allows for these 3 things to be ANDed together.